src/openepd/model/common.py

```diff
@@ -99,13 +99,14 @@ class WithAttachmentsMixin(pyd.BaseModel):
         default=None,
     )
 
-    def set_attachment(self, name: str, url: str):
+    def set_attachment(self, name: str, url: str) -> None:
         """Set an attachment."""
+        url = cast(pyd.AnyUrl, url)
```

Comment:
This does not look good; it can be string but not URL. In this case object will be stored, serialized, but won't be read back.